### PR TITLE
feat: Improve stage wizard discoverability and interaction COMPASS-7350

### PR DESCRIPTION
**Pull request by mcasimir**
Improves the stage wizard discoverability:

- Swap the trigger IconButton with a regular Button with an Icon and a label
- Make the panel open by default and remember its state
- Remove links from the card and use the badge instead
- Make it feel more clickable
- Change the copy in the search box
- Make the card dimmed while dragging
- Prioritize the "click" interaction (insert as last stage)
- Give back focus to the trigger after cancelling the form

https://github.com/mongodb-js/compass/assets/334881/0c5b50aa-4d2a-47f1-8e50-978392387086

Screenshot 2023-10-18 at 10 36 43

### CODE REVIEW

File: packages/compass-components/src/components/keyline-card.tsx [hunk 1]
```diff
@@ -17,23 +20,74 @@ const keylineDarkThemeStyles = css({
...
 
+const lightHoverBoxShadow = `0 4px 20px -4px ${transparentize(
+  0.8,
+  palette.black
+)}`;
+
+const darkHoverBoxShadow = `0 4px 20px -4px ${transparentize(
+  0.3,
+  palette.black
+)}`;
```
Comment (Anemy) [inline packages/compass-components/src/components/keyline-card.tsx:31]: Looking closely this hover seems to go out of the boundaries and gets a cut off look:
Screenshot 2023-10-19 at 11 28 16 AM
I see that this is the same as the [clickable card styles](https://github.com/mongodb/leafygreen-ui/blob/cf06988dcef8e2dac83ebc409f3b2e9a905fe9e8/packages/card/src/Card/styles.ts#L22), so I'm assuming we want to keep them.
Maybe we can make the container padding and margins set in the way where the items are not cut off?
Fine as is too, this is a design nit.

File: packages/compass-components/src/components/keyline-card.tsx [hunk 2]
```diff
@@ -17,23 +20,74 @@ const keylineDarkThemeStyles = css({
...
-  return (
-    <div
-      ref={ref}
-      className={cx(
+  const allProps = mergeProps(
+    contentStyle === 'clickable'
+      ? { role: 'button', tabIndex: 0, ...focusRingProps }
+      : {},
+    {
+      className: cx(
```
Comment (Anemy) [inline packages/compass-components/src/components/keyline-card.tsx:79]: Related to the above question on box shadows, the clickable `Cards` from leafygreen have a box shadow that's always showing, and expanded when hovered. Would that make this more `clickable` looking? Fine as is too I reckon.

| not hovered | hovered |
| - | - |
| Screenshot 2023-10-19 at 11 47 39 AM | Screenshot 2023-10-19 at 11 47 42 AM |

File: packages/compass-components/src/components/keyline-card.tsx [hunk 3]
```diff
@@ -17,23 +20,74 @@ const keylineDarkThemeStyles = css({
...
+  }
+`;
+
+const clickableStyleDark = css`
+  cursor: pointer;
+
+  &:hover {
+    box-shadow: ${darkHoverBoxShadow};
+  }
+`;
```
Comment (Anemy) [inline packages/compass-components/src/components/keyline-card.tsx:52]: To make the hover smoother, I'm thinking lets pull over the transition styles from lg:
https://github.com/mongodb/leafygreen-ui/blob/cf06988dcef8e2dac83ebc409f3b2e9a905fe9e8/packages/card/src/Card/styles.ts#L86C1-L87C43

```
import {
  transitionDuration
} from '@leafygreen-ui/tokens';
```


```suggestion
const clickableStyleLight = css`
  cursor: pointer;
  transition: ${transitionDuration.default}ms ease-in-out;
  transition-property: border, box-shadow;

  &:hover {
    border: 1px solid ${palette.gray.light2};
    box-shadow: ${lightHoverBoxShadow};
  }

  &:active {
    box-shadow: none;
  }
`;

const clickableStyleDark = css`
  cursor: pointer;
  transition: ${transitionDuration.default}ms ease-in-out;
  transition-property: border, box-shadow;

  &:hover {
    box-shadow: ${darkHoverBoxShadow};
  }
`;
```

A total nit, but we might want to standardize on using `css({...})` vs 'css`...`' in these places. Although this is pulled from LG that uses 'css`...`'.

**Review by Anemy — COMMENTED** [review]
Looking good, a couple styling comments, probably not blockers.

File: packages/compass-components/src/components/keyline-card.tsx (hunk 2 above)
Comment (mcasimir) [reply packages/compass-components/src/components/keyline-card.tsx:79]: mmm weird, I'm sure i copied the styles as they were in Card, I'll play around with that

File: packages/compass-components/src/components/keyline-card.tsx (hunk 3 above)
Comment (mcasimir) [reply packages/compass-components/src/components/keyline-card.tsx:52]: Sounds great!

File: packages/compass-aggregations/src/components/pipeline-toolbar/pipeline-settings/hidden-on-narrow-container.ts [hunk 4]
```diff
@@ -0,0 +1,8 @@
+import { css } from '@mongodb-js/compass-components';
+import { workspaceContainerQueryName } from '@mongodb-js/compass-components';
```
Comment (gribnoysup) [inline packages/compass-aggregations/src/components/pipeline-toolbar/pipeline-settings/hidden-on-narrow-container.ts:2]: I was thinking about what we discussed yesterday and to make it more contextual, how do you feel about assigning this property to the component constructor? Maybe a good way to make it super obvious what the container query refers to?

```js
import { WorkspaceContainer } from '@mongodb-js/compass-components';

css`
  @container ${WorkspaceContainer.containerQueryName} (...) { ... }
`
```

Super nit though, feel free to ignore, just thinking how to make it  not just a random-ish const export from the components package

Comment (mcasimir) [reply packages/compass-aggregations/src/components/pipeline-toolbar/pipeline-settings/hidden-on-narrow-container.ts:2]: Makes sense!

Actually, I was almost about to remove this from WorkspaceContainer :). While doing this I stumbled on a couple of controls for which, if we want to do something nice we may still need something for react, cause hiding labels with css on LG controls is not the easiest thing ever :(

File: packages/compass-components/src/components/keyline-card.tsx (hunk 2 above)
Comment (mcasimir) [reply packages/compass-components/src/components/keyline-card.tsx:88]: turns out i've ditched those styles by accident, added them back

File: packages/compass-components/src/components/keyline-card.tsx (hunk 3 above)
Comment (mcasimir) [reply packages/compass-components/src/components/keyline-card.tsx]: done

File: packages/compass-components/src/components/keyline-card.tsx (hunk 1 above)
Comment (mcasimir) [reply packages/compass-components/src/components/keyline-card.tsx:40]: probably done .. i moved the paddings around

File: packages/compass-aggregations/src/components/aggregation-side-panel/index.tsx [hunk 5]
```diff
@@ -132,8 +135,8 @@ export const AggregationSidePanel = ({
       <SearchInput
         value={searchText}
         onChange={handleSearchTextChange}
-        placeholder="How can we help?"
-        aria-label="How can we help?"
+        placeholder="Search a Stage Wizard"
+        aria-label="Search a Stage Wizard"
```
[the remark this answers is not in the document]
Comment (mcasimir) [reply packages/compass-aggregations/src/components/aggregation-side-panel/index.tsx]: ended up with Search for a Stage, under suggestion for claudia and julia

File: packages/compass-components/package.json [hunk 6]
```diff
@@ -34,6 +34,7 @@
     "@dnd-kit/core": "^6.0.7",
     "@dnd-kit/sortable": "^7.0.2",
     "@dnd-kit/utilities": "^3.2.1",
+    "@emotion/css": "^11.11.2",
```
Comment (Anemy) [inline packages/compass-components/package.json:37]: Why do we need this dependency?

**Review by Anemy — COMMENTED** [review]
Changes look good, just one question on the emotion dependency. Should we always be using the styles from leafygreen emotion?

File: packages/compass-components/package.json (hunk 6 above)
Comment (gribnoysup) [reply packages/compass-components/package.json:37]: [Context for this](https://mongodb.slack.com/archives/G2L10JAV7/p1698163251582629?thread_ts=1698151766.108829&cid=G2L10JAV7)

Comment (Anemy) [reply packages/compass-components/package.json:37]: Ah cool, thanks. I wish we could leave a comment in json for on this one. Would it help if we add a comment in the `depcheck.rc`?

**Review by Anemy — APPROVED** [review]
lgtm! looks good

File: packages/compass-components/package.json (hunk 6 above)
Comment (mcasimir) [reply packages/compass-components/package.json:37]: sounds good, i'll follow up with that in a different PR, i have more random improvements coming up

**Pull request merged**